packages/core/src/components/navigation/NavbarLinks/NavbarLinks.tsx

```diff
@@ -26,18 +32,28 @@ function NavbarLinks({
 }: NavbarLinksProps) {
   const { NavbarLinks: NavbarLinksWrapper, NavbarLinksList } =
     useOverrideComponents<'Navbar'>()
+  const router = useRouter()
+  const { postalCode } = useSession()
+  const { filterByPickupPoint } = getRegionalizationSettings()
+
+  const shouldDisplayGlobalFilter =
+    deliveryPromise.enabled && postalCode && filterByPickupPoint?.enabled
+
   return (
     <NavbarLinksWrapper.Component {...otherProps} {...NavbarLinksWrapper.props}>
       <div data-fs-navbar-links-wrapper data-fs-content="navbar">
         {shouldDisplayRegion && (
           <RegionButton icon={regionIcon} label={regionLabel} />
         )}
+        {shouldDisplayGlobalFilter && (
```

Comment:
```suggestion
        {shouldDisplayGlobalFilter && filterByPickupPoint?.label && (
```
I think we need to add another conditional here. The button is still being rendered even with label undefined.. 
<img width="1531" alt="image" src="https://github.com/user-attachments/assets/64310709-52f5-483a-b0fb-a954ce340003" />

Reply:
I think it was happening because the `shouldDisplayGlobalFilter` was being set as different type values, not boolean. I fixed it in the last commit, test again to see if it happens.

Reply:
I've tried again ✅ it's fixed! Niceee!
<img width="900" alt="image" src="https://github.com/user-attachments/assets/806fe907-abb0-435f-b374-a3e0c1bd4157" />